diff --git a/webapi_core2/Models/IteminVideosforUser.cs b/webapi_core2/Models/IteminVideosforUser.cs
index 42e6d99..47eb07f 100644
--- a/webapi_core2/Models/IteminVideosforUser.cs
+++ b/webapi_core2/Models/IteminVideosforUser.cs
@@ -1,11 +1,14 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace webapi_core2.Models
 {
     public partial class IteminVideosforUser
     {
+        [Key]
         public int UserId { get; set; }
+        [Key]
         public int VideoId { get; set; }
 
         public User User { get; set; }
diff --git a/webapi_core2/Models/VoteforVideo.cs b/webapi_core2/Models/VoteforVideo.cs
index 7dc9332..c9474a6 100644
--- a/webapi_core2/Models/VoteforVideo.cs
+++ b/webapi_core2/Models/VoteforVideo.cs
@@ -1,11 +1,14 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace webapi_core2.Models
 {
     public partial class VoteforVideo
     {
+        [Key]
         public int VideoId { get; set; }
+        [Key]
         public int UserId { get; set; }
         public int? Mark { get; set; }
 
